techmatters/hrm
# feat: add updatedBy and createdBy columns to profiles tables [CHI-2603]

## Description
This PR:
- Adds `createdBy` and `updatedBy` columns to the profiles related tables.
- Triggers an update on profiles table when a profile flag is associated or disassociated.
- Triggers an update on profiles table when a profile is associated to an identifier.

`Profiles`
- after creation
  ![Screenshot from 2024-02-22 12-23-24](https://github.com/techmatters/hrm/assets/15805319/8d3c1d30-0a45-41ef-ab7d-55b4544058df)
- after associating a flag
  ![Screenshot from 2024-02-22 12-24-33](https://github.com/techmatters/hrm/assets/15805319/c83a21ff-ef39-4c01-8746-443756bda68e)

`ProfileSections`
- after creation
  ![Screenshot from 2024-02-22 12-26-06](https://github.com/techmatters/hrm/assets/15805319/c4134ede-6a07-4722-b667-fed92f7c61fd)
- after editing
  ![Screenshot from 2024-02-22 12-26-25](https://github.com/techmatters/hrm/assets/15805319/bbbc2e42-a98f-4338-8719-90d91536d256)

Same goes for flags and identifiers, but since those two can only be created/updated by system or Aselo staff, I'm skipping the screenshots.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2603)

### Verification steps
Trust service tests or
- Run local server (hrm and Flex).
- Create a new contact from a profile that does not exists in your DB.
  - [ ] Confirm that the profile and the identifier are created under `system` user.
- Associate a new flag to the profile
  - [ ] Confirm that the profile captures the `updatedBy` matching the user who associated.
- Disssociate a flag from the profile
  - [ ] Confirm that the profile captures the `updatedBy` matching the user who disassociated.
- Create a new profile section
  - [ ] Confirm that the section captures the `createdBy` matching the user.
- Edit a new profile section
  - [ ] Confirm that the section captures the `updatedBy` matching the user.
